zeppelin-server/src/main/java/org/apache/zeppelin/rest/message/NewNoteRequest.java

```diff
@@ -16,30 +16,33 @@
  */
 package org.apache.zeppelin.rest.message;
 
-import com.google.gson.Gson;
-
 import java.util.List;
 
-import org.apache.zeppelin.common.JsonSerializable;
-
 /**
  *  NewNoteRequest rest api request message.
  */
-public class NewNoteRequest implements JsonSerializable {
-  private static final Gson GSON = new Gson();
+public class NewNoteRequest {
 
   //TODO(zjffdu) rename it to be notePath instead of name
-  private String name;
-  private String defaultInterpreterGroup;
-  private boolean addingEmptyParagraph = false;
-  private List<NewParagraphRequest> paragraphs;
-  private String revisionId;
+  private final String name;
+  private final String defaultInterpreterGroup;
+  private final Boolean addingEmptyParagraph;
```

Comment:
If the parameter is missing in the request and GSON deserializes a primitive data type, then the value cannot be null. Here it is a boolean, the default here is probably false.
Unfortunately I did not find any GSON documentation of the default values of the primitive datatypes. So I decided to use the class Boolean to keep the default value with the null check.
If you have a better idea, I would be very grateful. It should be clear what the default value is.

Reply:
It looks quite complicated. I understood your intention.

Reply:
Do you have any idea how I can improve this. The `final` keyword on the `Boolean` is quite important to me, so that it is clear that you cannot edit the object from the request.

Reply:
I'll also check it how to improve it.

Reply:
I found the part of the documentation that describes the default value for boolean and other primitive types.
https://github.com/google/gson/blob/master/UserGuide.md#finer-points-with-objects

After consulting with the developers in my company, I will look for primitive data types in other API endpoints and add null checking here as well.